### PR TITLE
Add support for single-end fastqs

### PR DESCRIPTION
**Pull request by tfrcarvalho**
### Description

* Support single-end fastqs (in addition to paired-end): [CH-18084](https://app.clubhouse.io/idseq/story/18084)
* Fixes bug where the FilterReads step would error earlier with no reads left without reporting the correct error
* Fixes bug where non gzip input fastqs would originated 0 total reads
* Improves precision of quantifiers for `Array[File]` outputs

### Tests

* Tested locally with:
  * Paired-end Gzip Fastq with no reads left after in FilterReads task
  * Single-end Gzip Fastq with consensus genome created
  * Paired-end Gzip Fastq with consensus genome created (same output as single-end)
  * Paired-end uncompressed Fastq with consensus genome created

### CODE REVIEW

File: consensus-genome/run.wdl [hunk 1]
```diff
@@ -139,41 +133,38 @@ workflow consensus_genome {
...
-                Quast.quast_tsv,
-                AlignReads.alignments,
-                QuantifyERCCs.ercc_out,
-                QuantifyERCCs.ercc_out,
-                ComputeStats.output_stats,
-                CallVariants.variants_ch
-            ]),
+            outputFiles = select_all(flatten([
+                RemoveHost.host_removed_fastqs,
+                select_all([
```
Comment (jshoe) [inline consensus-genome/run.wdl:146]: just curious, do you need the additional `select_all` inside if you're flattening the whole?

**Review by jshoe — APPROVED** [review]
LGTM with a question

File: consensus-genome/run.wdl (hunk 1 above)
Comment (tfrcarvalho) [reply consensus-genome/run.wdl:146]: This was a whole fight with miniwdl type checking... so, yes, but i not 100% sure on an explanation.
Might follow up on that with @mlin later.

Comment (tfrcarvalho) [reply consensus-genome/run.wdl:146]: I believe it was something to do with some of those files being optional...

**Pull request merged**